src/screens/token/components/account-settings/account-settings.tsx

```diff
@@ -49,33 +49,43 @@ export class AccountSettingsComponent extends React.Component<IProps & IExternal
             title: translate('AccountSettings.manageAccount'),
             key: '',
             showSecurityWarning: false,
-            isLoading: false
+            isLoading: false,
+            isModalVisible: props.isVisible
         };
     }
 
-    public async revealPrivateKey() {
+    public componentDidUpdate(prevProps: IExternalProps) {
+        if (this.props.isVisible !== prevProps.isVisible) {
```

Comment:
we can avoid using lifecycle function in this case. 
- keep the visibility state of account options modal in props
- in state add the visibility state of other modal
Example:

```ts
interface IState {
  ...
  displayOtherModal: boolean;
}
...
render() {
...
  <Modal isVisible={this.props.visible && !this.state.displayOtherModal}>
...
}
```